### DISCUSSION TITLE
error: failed to run custom build command for `ring v0.16.20`

### DISCUSSION THREAD

**Discussion by SsebowaDisan** [shadowsocks/shadowsocks-android#2802] [Q&A]
**Unable to compile shadowsocks-android project. It gives me an failed to run command ring**
Compiling on Ubuntu: Ubuntu 18.04.5 LTS
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `/home/stemeye/Desktop/vpn/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/debug/build/ring-bbabc15ac3b86348/build-script-build` (exit status: 101)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("armv7-linux-androideabi")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_armv7-linux-androideabi = None
  CC_armv7_linux_androideabi = None
  TARGET_CC = None
  CC = None
  CFLAGS_armv7-linux-androideabi = None
  CFLAGS_armv7_linux_androideabi = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")

  --- stderr
  running "arm-linux-androideabi-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/stemeye/Desktop/vpn/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/armv7-linux-androideabi/debug/build/ring-c9db317f4f72d1f9/out/aesv8-armx-linux32.o" "/home/stemeye/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux32.S"
  thread 'main' panicked at 'failed to execute ["arm-linux-androideabi-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/stemeye/Desktop/vpn/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/armv7-linux-androideabi/debug/build/ring-c9db317f4f72d1f9/out/aesv8-armx-linux32.o" "/home/stemeye/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux32.S"]: No such file or directory (os error 2)', /home/stemeye/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/build.rs:653:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
failed to run custom build command for `ring v0.16.20`

Caused by:

build failed, waiting for other jobs to finish...

error: build failed

> Task :core:cargoBuildArm FAILED
build failed

**Comment by IkramulMurad** [comment]
> **Unable to compile shadowsocks-android project. It gives me an failed to run command ring** Compiling on Ubuntu: Ubuntu 18.04.5 LTS error: failed to run custom build command for `ring v0.16.20`
> 
> Caused by: process didn't exit successfully: `/home/stemeye/Desktop/vpn/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/debug/build/ring-bbabc15ac3b86348/build-script-build` (exit status: 101) --- stdout OPT_LEVEL = Some("0") TARGET = Some("armv7-linux-androideabi") HOST = Some("x86_64-unknown-linux-gnu") CC_armv7-linux-androideabi = None CC_armv7_linux_androideabi = None TARGET_CC = None CC = None CFLAGS_armv7-linux-androideabi = None CFLAGS_armv7_linux_androideabi = None TARGET_CFLAGS = None CFLAGS = None CRATE_CC_NO_DEFAULTS = None DEBUG = Some("true")
> 
> --- stderr running "arm-linux-androideabi-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/stemeye/Desktop/vpn/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/armv7-linux-androideabi/debug/build/ring-c9db317f4f72d1f9/out/aesv8-armx-linux32.o" "/home/stemeye/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux32.S" thread 'main' panicked at 'failed to execute ["arm-linux-androideabi-clang" "-O0" "-DANDROID" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-I" "include" "-Wall" "-Wextra" "-pedantic" "-pedantic-errors" "-Wall" "-Wextra" "-Wcast-align" "-Wcast-qual" "-Wconversion" "-Wenum-compare" "-Wfloat-equal" "-Wformat=2" "-Winline" "-Winvalid-pch" "-Wmissing-field-initializers" "-Wmissing-include-dirs" "-Wredundant-decls" "-Wshadow" "-Wsign-compare" "-Wsign-conversion" "-Wundef" "-Wuninitialized" "-Wwrite-strings" "-fno-strict-aliasing" "-fvisibility=hidden" "-fstack-protector" "-g3" "-DNDEBUG" "-c" "-o/home/stemeye/Desktop/vpn/shadowsocks-android/core/src/main/rust/shadowsocks-rust/target/armv7-linux-androideabi/debug/build/ring-c9db317f4f72d1f9/out/aesv8-armx-linux32.o" "/home/stemeye/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/pregenerated/aesv8-armx-linux32.S"]: No such file or directory (os error 2)', /home/stemeye/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/build.rs:653:9 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace warning: build failed, waiting for other jobs to finish... failed to run custom build command for `ring v0.16.20`
> 
> Caused by:
> 
> build failed, waiting for other jobs to finish...
> 
> error: build failed
> 
> > Task :core:cargoBuildArm FAILED
> > build failed

I'm also getting this error.
Have you solved this error?
If so, could you please say?
